Various makefile cleanups, lite edition

josuah opened the pull request:
This pull request brings some of the Makefile cleanups from #1892:

* Improvements for Makefile.vcs (from @imphil)
* Fix a display bug with `$(error ...)` (from @imphil)
* `rm -f` being wrapped-up in an `$(RM)` macro (from @imphil)
* `make` being wraped-up in a `$(MAKE)` macro
* `find | xargs rm` being replaced in `find -exec rm {} +`, covering very rare corner-cases. I like piping though.

The `COCOTB_CONFIG_CMD` changes were not included, for which a dedicated branch can be made.

josuah commented:
I feel a bit like taking the work of @imphil and putting my name on it, but the branches diverged and I could not `git format-patch | git am` the changes.

ktbarrett commented on Makefile:
```diff
@@ -32,9 +32,9 @@ all: test
-	-@find . -name "obj" | xargs rm -rf
-	-@find . -name "*.pyc" | xargs rm -rf
-	-@find . -name "*results.xml" | xargs rm -rf
+	-@find . -name "obj" -exec rm -rf {} +
+	-@find . -name "*.pyc" -exec rm -rf {} +
+	-@find . -name "*results.xml" -exec rm -rf {} +
```
There is also the `-delete` option to `find`.

ktbarrett commented on cocotb/share/makefiles/simulators/Makefile.activehdl:
```diff
@@ -59,14 +59,14 @@ endif
-	@rm -rf $(SIM_BUILD)
-	@rm -rf work
-	@rm -rf wave.asdb
+	$(RM) $(SIM_BUILD)
```
?
```suggestion
	$(RM) -r $(SIM_BUILD)
```

ktbarrett commented on cocotb/share/makefiles/simulators/Makefile.riviera:
```diff
@@ -143,15 +143,15 @@ endif
-	@rm -rf compile
-	@rm -rf library.cfg
-	@rm -rf dataset.asdb
+	$(RM) $(SIM_BUILD)
```
?
```suggestion
	$(RM) -r $(SIM_BUILD)
```

ktbarrett reviewed (commented):
> I feel a bit like taking the work of @imphil and putting my name on it

You can rewrite your commits and add him as a coauthor. See [here](https://github.blog/2018-01-29-commit-together-with-co-authors/) for some more details.

josuah commented:
I adjusted the Makefiles as you suggested. I forgot that I did not add the `-r` in the Makefile.
It might be better to let the GNU make default `rm -f` apply (I removed the `RM :=`).

ktbarrett commented on Makefile:
```diff
@@ -32,9 +32,9 @@ all: test
-	-@find . -name "obj" | xargs rm -rf
-	-@find . -name "*.pyc" | xargs rm -rf
-	-@find . -name "*results.xml" | xargs rm -rf
+	-@find . -name "obj" -exec rm -rf {} +
```
Huh, `-delete` doesn't work on directories. I love it when flags don't behave consistently. There doesn't seem to be a `-deletedir` option either. No action, just leaving justification for posterity.

josuah replied:
Now that I think of it `find . -depth \( -path "*/obj/*" -o -name "obj" \) -delete` would do it, but well...

I could not find "obj" in the hierarchy, this lines comes from the original commit, and I am not sure if it is a file or a directory.

ktbarrett replied:
I'm not sure either. Better to just leave it be.

imphil reviewed (approved):
Thanks a lot for picking up this work, @josuah, I totally forgot that I had almost-finished changed around!
